app/src/main/java/com/communistutopia/spacetrader/model/Inventory.kt

```diff
@@ -16,4 +16,33 @@ package com.communistutopia.spacetrader.model
  */
 class Inventory(
     var water: Int, var furs: Int, var food: Int, var ore: Int, var games: Int, var firearms: Int,
-                var medicine: Int, var machines: Int, var narcotics: Int, var robots: Int)
\ No newline at end of file
+                var medicine: Int, var machines: Int, var narcotics: Int, var robots: Int) {
+
+
+    private var supplies: MutableMap<String, Int>
+
+    init {
+        //TODO: Refactor this. It's real bad hardcoding but should be acceptable for the purposes of M7 demo
+        supplies = mutableMapOf("water" to water, "furs" to furs, "food" to food, "ore" to ore, "games" to games, "firearms" to firearms,
+                        "medicine" to medicine, "machines" to machines, "narcotics" to narcotics, "robots" to robots)
+    }
+
+    fun getValue(good: TradeGood): Int {
+        return supplies[good.name]!!
```

Comment:
It force unwraps the optional value. Kotlin dicts are a little weird, but it won't allow you to return if it can be null. this operator forces that value to not be null. this is enforced in the Inventory initialization (Hard-coded for now. Technical debt reacts only)